### PR TITLE
Widescreen 2: The Electric Boogaloo

### PR DESCRIPTION
**Pull request by 567Turtle**
# About the pull request

All the features of Widescreen 1, but functional!

# Explain why it's good for the game

Widescreen (and diagonal movement but that's another can of beans) are necessary features that most other codebases have already implemented. I'm a subscriber to the ideology that modernization of SS13 is a good thing, instead of staying stuck in the past. 

I believe a change such as this will have very little effect to the balance of the game, as scopes and bug range has been changed accordingly. It WILL however make the game easier to get into for new players, as well as making the game just easier to look at overall as the screen will no longer be 50% text box.

<details>
<summary>Screenshots & Videos</summary>

prof

this is my minecraft skin btw what do you guys think
tuuefiing1

</details>

datumised view refactor courtesy of https://github.com/tgstation/TerraGov-Marine-Corps/pull/6748

# Changelog
:cl: TiviPlus, HarryOB, 567Turtle
add: Widescreen preference :)
/:cl:

### CODE REVIEW

**Comment by 567Turtle** [comment]
> I'm not a fan of being able to see way further in certain directions than in others. Always seemed to me like an LRP thing ruining immersion.

I'm all for immersion and roleplay, it's one of my favorite aspects of CM believe it or not, and not once did it cross my mind that it would break immersion. I really just don't see this as an issue at all.

**Comment by Vicacrov** [comment]
The "Hive Strengthening..." message is no longer centralised, not sure I should be reporting it here or as an issue report to be fixed later.

image

**Comment by Reisyns** [comment]
Jarring as fuck to play with and the reasoning of it being so that 50% of your screen isn't taken up by a text box is invalid as you can just change the size of the right window at the same time as centring your screen more. Just because you want to modernize something doesn't mean it'll be a better change, some things don't need to be changed. But some people don't get that.

**Comment by sl0w3st** [comment]
Isn't text box very vital part of the game that you need to take your attention to?

Also, what about users with different monitor sizes?

File: code/datums/view.dm [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+/datum/view_data/proc/get_client_view_size()
+	var/list/temp = getviewsize(default)
+	if(suppress_changes)
+		return "[temp[1]]x[temp[2]]"
+	return "[width + temp[1]]x[height + temp[2]]"
+
+///Zooms the client back in with an animate pretty simple
+/datum/view_data/proc/zoom_in()
+	reset_to_default()
+	animate(parent, pixel_x = 0, pixel_y = 0, 0, FALSE, LINEAR_EASING, ANIMATION_END_NOW)
```
Comment (TheGamerdk) [inline code/datums/view.dm:126]: Just a useability comment, but if this is what's causing the smooth transition when leaving binoculars, please, for the love of God, make it a toggle or something. It's motion sickness inducing.

Comment (TheGamerdk) [reply code/datums/view.dm:126]: it wasn't nevermind, thanks harry

**Comment by Tsurupeta** [comment]
If you have square monitor, the already small textbox will become even smaller.

**Comment by sl0w3st** [comment]
It looks like NVG users got free x2 scope and globber boiler has bigger risks at bombarding to east or west directions

**Comment by sl0w3st** [comment]
You can't point at things nor zoom your screen as observer

**Comment by sl0w3st** [comment]
GL spec and anyone wielding M79 got straight buffed with this thing

**Comment by GrrrKitten** [comment]
Im crashing a lot from this when I use it with fullscreen mode. I have fullscreen mode enabled by default but I still have to unfullscreen and re-fullscreen pretty much everytime I get in. clearing cache and the usual crash fixers has not been helpful

**Comment by Mister-moon1** [comment]
Issue regarding the viewport automatically resetting regardless of if you have it set to manual

**Comment by 567Turtle** [comment]
image

**Comment by Staykeu** [comment]
bad take

**Comment by GrrrKitten** [comment]
it took me two rounds for it to feel pretty natural, I think the balance impacts of it would have even'd out over time. even though I am still not stold on widescreen for CM, I think this is giving up kinda early on the instant negative reaction that all PR's that change anything get.

Especially when things like dropoff can be adjusted.

(also if youre truly giving up on this pls let us get psuedo-widescreen via moving the hud to the right side of the viewport here xooxooxoxo 
image)

**Comment by Steelpoint** [comment]
A shame, I liked it.

**Pull request closed without merging**